runtime/src/iree/hal/drivers/hip/hip_driver.c

```diff
@@ -465,10 +472,120 @@ static iree_status_t iree_hal_hip_driver_create_device_by_index(
   IREE_HIP_RETURN_IF_ERROR(&driver->hip_symbols,
                            hipDeviceGet(&device, device_index), "hipDeviceGet");
 
-  iree_status_t status = iree_hal_hip_driver_create_device_by_id(
-      base_driver, IREE_HIPDEVICE_TO_DEVICE_ID(device), param_count, params,
-      host_allocator, out_device);
+  *out_id = IREE_HIPDEVICE_TO_DEVICE_ID(device);
+  return iree_ok_status();
+}
 
+static iree_status_t iree_hal_hip_driver_get_device_id_by_path(
+    iree_hal_driver_t* base_driver, iree_string_view_t device_path,
+    iree_hal_device_id_t* out_id) {
+  iree_hal_hip_driver_t* driver = iree_hal_hip_driver_cast(base_driver);
+  IREE_RETURN_IF_ERROR(iree_hal_hip_init(driver));
+
+  iree_hal_device_id_t device_id = 0;
+  iree_status_t status = iree_hal_hip_driver_get_device_id_by_uuid(
+      base_driver, device_path, &device_id);
+  if (iree_status_is_ok(status)) {
+    *out_id = device_id;
+    return iree_ok_status();
+  }
+  if (!iree_status_is_invalid_argument(status)) {
+    return status;
+  }
+  status = iree_hal_hip_driver_get_device_id_by_index(base_driver, device_path,
+                                                      &device_id);
+  if (iree_status_is_ok(status)) {
+    *out_id = device_id;
+    return iree_ok_status();
+  }
+  if (!iree_status_is_invalid_argument(status)) {
+    return status;
+  }
+  return iree_make_status(IREE_STATUS_UNIMPLEMENTED, "unsupported device path");
```

Comment:
This can be simplified - the function as a whole makes it hard to know what's happening and the convention we use is useful to avoid that.

This should be something like
```c
if (iree_status_is_ok(status)) {
  *out_id = device_id;
}
return status;
```

You should have verified/set the error above and just be handling the function result here. Instead of returning iree_ok_status() after an iree_status_is_ok prefer just passing it through - keeps the code smaller and avoids extra cognitive load for readers.